syntax = "proto3";

package userinfo;

message LoginRequest
{
    string username = 1;
    string password = 2;
}

message LoginResponse
{
    int32 ret_code = 1;
    string ret_msg = 2;
    string token = 3;
}

message RegisterRequest
{
    string username = 1;
    string password = 2;
}

message RegisterResponse
{
    int32 ret_code = 1;
    string ret_msg = 2;
    string token = 3;
}

message TestRequest
{
    string token = 1;
    string test_string = 2;
}

message TestResponse
{
    int32 ret_code = 1;
    string ret_msg = 2;
    string ret_str = 3;
}

service UserAction
{
    rpc Login(LoginRequest) returns (LoginResponse) {}
    rpc Register(RegisterRequest) returns (RegisterResponse) {}
    rpc Test(TestRequest) returns (TestResponse) {}
}
